src/main/java/com/microsoft/graph/http/CoreHttpProvider.java

```diff
@@ -236,8 +242,14 @@ public ISerializer getSerializer() {
 		logger.logDebug("Starting to send request, URL " + requestUrl.toString());
 
 		// Request level middleware options
-		final RedirectOptions redirectOptions = request.getMaxRedirects() <= 0 ? null : new RedirectOptions(request.getMaxRedirects(), request.getShouldRedirect());
-		final RetryOptions retryOptions = request.getShouldRetry() == null ? null : new RetryOptions(request.getShouldRetry(), request.getMaxRetries(), request.getDelay());
+		final RedirectOptions redirectOptions =
+            request.getMaxRedirects() == DEFAULT_MAX_REDIRECTS && request.getShouldRedirect().equals(DEFAULT_SHOULD_REDIRECT)
+                ? null
+                : new RedirectOptions(request.getMaxRedirects(), request.getShouldRedirect());
+		final RetryOptions retryOptions =
+            request.getMaxRetries() == DEFAULT_MAX_RETRIES && request.getDelay() == DEFAULT_DELAY && request.getShouldRetry().equals(DEFAULT_SHOULD_RETRY)
+                ? null
+                : new RetryOptions(request.getShouldRetry(), request.getMaxRetries(), request.getDelay());
```

Comment:
Is this required for giving the flexibility to custom handler implementations?

In the default implementation of `RetryHandler` and `RedirectHandler`, the constructors end up calling the default option constructors if they are `null`. Default constructors of options are already using these default values. This piece of code seems equivalent to:
```suggestion
		final RedirectOptions redirectOptions = new RedirectOptions(request.getMaxRedirects(), request.getShouldRedirect());
		final RetryOptions retryOptions = new RetryOptions(request.getShouldRetry(), request.getMaxRetries(), request.getDelay());
```

Reply:
Hi @zengin
The changes you are suggesting work effectively the same as the current code of the library. If you take a look at **CoreHttpProvider's** line 245:
```java
if(redirectOptions != null) {
	corehttpRequestBuilder = corehttpRequestBuilder.tag(RedirectOptions.class, redirectOptions);
}
if(retryOptions != null) {
	corehttpRequestBuilder = corehttpRequestBuilder.tag(RetryOptions.class, retryOptions);
}
```
**RedirectHandler's** line 133:
```java
RedirectOptions redirectOptions = request.tag(RedirectOptions.class);
redirectOptions = redirectOptions != null ? redirectOptions : this.mRedirectOptions;
```
and **RetryHandler's** line 179:
```java
RetryOptions retryOption = request.tag(RetryOptions.class);
retryOption = retryOption != null ? retryOption : mRetryOption;
```
you can see that **mRedirectOptions** and **mRetryOption** will never be used. Thus it is impossible to customize them so to speak globaly.

Reply:
Thanks for clarifying.